### PR TITLE
Split Save Icon Data and Refactored code for it.

### PR DESCRIPTION
**Pull request by smb123w64gb**

### CODE REVIEW

File: src/dra/42398.c [hunk 1]
```diff
@@ -834,29 +834,17 @@ s32 func_800E9B18(s32 arg0, s32 arg1) {
...
     for (i = 0; i < 16; i++) {
-        *arg0 = *var_v1;
-        var_v1++;
-        arg0++;
+        PaletteDestination[i] = SaveIconPalette[SelectedPalette][i];
     }
 }
 
-void func_800E9BDC(u8* arg0, s32 arg1) {
+void Load_Save_Icon(u8* IconDestination, s32 SelectedIcon) {
```
Comment (sonicdcer) [inline src/dra/42398.c:844]: void loadSaveIcon(u8* iconDestination, s32 selectedIcon) {

**Comment by Xeeynamo** [comment]
I am merging `master` into your PR as it fixes the `Validate decompiled code` step, so we can verify that the PR gives a green build

**Comment by Xeeynamo** [comment]
The code you suggested is 1:1 with the original compiled code. Congrats :clap: I am merging as it is much more readable now!

**Comment by sonicdcer** [comment]
I had to roll back your PR because your functions don't match. Can you reach out at Discord?
We have a small home at CV Romhacking
#castlevania-symphony-of-the-night
https://discord.gg/jEDZYkyYRY

**Pull request merged**